Makefile

```diff
@@ -8,13 +8,16 @@ ECHO1 = $(V:1=@ :)
 ECHO = $(ECHO1:0=@ echo)
 FUZZ_OUTPUT_DIR = $(shell pwd)/fuzz/output
 
-SOEXT := $(shell ruby -e 'puts RbConfig::CONFIG["SOEXT"]')
+SOEXT ?= $(shell ruby -e 'puts RbConfig::CONFIG["SOEXT"]')
```

Comment:
C extensions actually use `DLEXT`. But, considering this is the `Makefile` for the Prism core lib, _not_ the Prism extension gem, I kept it as `SOEXT`.

Both `DLEXT` and `SOEXT` are `so` on Linux Ruby, but Windows RubyInstaller has `so` for `DLEXT` – matching Linux’s – and `dll` for `SOEXT` – matching Windows conventions.

Reply:
Would you suggest changing this? I'm not familiar enough with windows dev to know if it would break something.

Reply:
I briefly reviewed `extconf.rb` and would recomment _against_ changing this, at least in this PR for scope reasons.

I have experimented before.
Extension Gems use `so` even on Windows because, appearently, the Ruby ecosystem _expects `.so` and ignores `.dll`_.
But OTOH, MinGW `-l` expects `.dll` to be the implicit file extension _but not `.so`_.

__TL;DR: Changing file extensions can break build pipelines.__

Reply:
On macOS one cannot link a .bundle against another .bundle AFAIK, so SOEXT is correct for libprism.